### PR TITLE
add an overview and skeleton docs for major sections

### PR DESCRIPTION
**Pull request by elmiko**

### CODE REVIEW

**Comment by elmiko** [comment]
this is not an authoritative list, it will need to be expanded as we add more information. the first thing that stands out to me is that we need a section for the installer changes.

File: overview.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    [OpenShift enhancements](https://github.com/openshift/enhancements).
+
+1. [Cloud Controller Manager](/cloud-controller-manager.md)
+
+    New providers must have a
+    [Cloud Controller Manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/)
+    integrated into OpenShift, including a repository in the OpenShift
+    organization and container images included in the release payload.
+
+1. [Machine API Controllers](/machine-api-controllers.md)
```
Comment (lobziik) [inline overview.md:35]: Just thought - maybe swap this with csi section?

File: overview.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    the [Machine API Operator](https://github.com/openshift/machine-api-operator),
+    including a repository in the OpenShift organization and container images
+    included in the release payload.
+
+1. [Container Storage Interface Driver](/container-storage-interface-driver.md)
+
+    Providers that expose storage options must have a CSI driver integrated into
+    OpenShift, including a repository in the OpenShift organization and container
+    images inclujded in the release payload.
+
```
Comment (lobziik) [inline overview.md:47]: Would suggest to add another section - smth like `Other Components`. For have a place to put info about things like image registry

**Comment by lobziik** [comment]
Overall looks great to me. Think it's a good starting point. Suggest to merge and proceed from there :)

File: overview.md (hunk 1 above)
Comment (elmiko) [reply overview.md:35]: just to be clear, would you want to see this section removed or place a csi section before it?

i don't think we should remove it, but i could see adding a csi section before it, just curious what you were thinking.

File: overview.md (hunk 2 above)
Comment (elmiko) [reply overview.md:47]: that sounds nice, i was reading this [old doc](https://github.com/smarterclayton/installer/blob/974d6cc015718028f74e38010cb8805fd35303cc/docs/dev/adding-new-platform.md) from Calyton's repo, and i like the distinction he draws between the "core platform" and "component infrastructure" parts.

File: overview.md (hunk 1 above)
Comment (lobziik) [reply overview.md:35]: Just place CSI before it.

**Comment by elmiko** [comment]
update
* added a networking section
* moved csi section
* added component infrastructure section

**Pull request merged**